plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -172,26 +175,33 @@ public ElasticsearchTableHandle getTableHandle(ConnectorSession session, SchemaT
                 // TODO this query pass-through mechanism is deprecated in favor of the `raw_query` table function.
                 //  it should be eventually removed: https://github.com/trinodb/trino/issues/13050
                 if (table.endsWith(PASSTHROUGH_QUERY_SUFFIX)) {
-                    table = table.substring(0, table.length() - PASSTHROUGH_QUERY_SUFFIX.length());
-                    byte[] decoded;
-                    try {
-                        decoded = BaseEncoding.base32().decode(parts[1].toUpperCase(ENGLISH));
-                    }
-                    catch (IllegalArgumentException e) {
-                        throw new TrinoException(INVALID_ARGUMENTS, format("Elasticsearch query for '%s' is not base32-encoded correctly", table), e);
-                    }
+                    if (this.legacyPassThroughQueryEnabled) {
```

Comment:
please invert the condition then you can return fast and avoid indentation. You will avoid changing lines. Easier to review ;)

```
if (!constraint) {
  throw exception
}
the old code
```